lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -178,6 +178,51 @@ class DecomposeAten_SoftmaxBackwardDataOp
 };
 } // namespace
 
+// Aten_LogSoftmaxBackwardDataOp(gradOutput, output, dim) =>
+//    result = gradOutput - (exp(output) * sum(gradOutput, dim))
+namespace {
+class DecomposeAten_LogSoftmaxBackwardDataOp
```

Comment:
`LogSoftmaxBackward` and `SoftmaxBackward` seem to overlap quite a bit. This should be refactored so that both ops share the common parts. Remember to avoid copying and pasting code if possible.

Reply:
Yes, thanks.

Reply:
I don't think this was fully addressed. In my opinion, the number of lines of code shared between `LogSoftmaxBackward` and `SoftmaxBackward` justifies a helper function. You should be able to create a helper function `decomposeSoftmaxBackwardDataLikeOp(Operation *op, PatternRewriter &rewriter)` that performs `return x - y * sum(x, dim)`, (assigning `x` and `y` accordingly depending on the `op`) so both `LogSoftmaxBackward` and `SoftmaxBackward` can share almost entirely all the code. This can make debugging a lot easier if a bug is found in the future, as well as making it easier to add features if needed.

Reply:
I am really sorry for this. I will make sure that this gets merged. I am spinning up a PR.

Reply:
@ramiro050  PR here: https://github.com/llvm/torch-mlir/pull/428